File: 2024/Leetcode/problem287.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        fast = nums[nums[0]]
        slow = nums[0]
        slow2 = 0

        while fast != slow:
            fast = nums[nums[fast]]
            slow = nums[slow]

        while slow2 != slow:
            slow2 = nums[slow2]
            slow = nums[slow]

        return slow

File: 2024/Leetcode/problem287.ts
function findDuplicate(nums: number[]): number {
    let slow = nums[0];
    let fast = nums[slow];
    let slow2 = 0;

    while (slow != fast) {
        slow = nums[slow];
        fast = nums[nums[fast]];
    }

    while (slow2 != slow) {
        slow = nums[slow];
        slow2 = nums[slow2];
    }

    return slow;
};